client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCacheManager.java

```diff
@@ -55,7 +55,7 @@
  * @author Mircea.Markus@jboss.com
  * @since 4.1
  */
-public class RemoteCacheManager implements BasicCacheContainer {
```

Comment:
Changing an class to become an interface would be problematic for existing users since that'd create a binary compatibility issue which we can work around it. So, although we're free to do the change you suggest, you'd make it more difficult for users to upgrade. So -1 from me.